ginivision/src/main/java/net/gini/android/vision/internal/camera/photo/Exif.java

```diff
@@ -75,20 +91,41 @@ public static RequiredTags readRequiredTags(@NonNull byte[] jpeg) throws IOExcep
         private TiffOutputDirectory mIfd0Directory;
         private TiffOutputDirectory mExifDirectory;
 
-        private Builder() throws ImageWriteException {
+        private Builder(@NonNull final byte[] jpeg)
+                throws ImageWriteException, IOException, ImageReadException {
             // Create a new exif metadata set, to keep only the required exif tags
-            mTiffOutputSet = new TiffOutputSet();
+            mTiffOutputSet = createOutputSetForJpeg(jpeg, TiffConstants.DEFAULT_TIFF_BYTE_ORDER);
```

Comment:
This change is important as it creates a TiffOutputDirectory with the byte order from the existing one in the jpeg if available. Uses the default byte order otherwise.